### PR TITLE
update tailwind and eslint config to be compatible with Next.js App Router, `@theguild/components` v8 and Nextra 4

### PR DESCRIPTION
**Pull request by dimaMachina**

### CODE REVIEW

File: packages/tailwind-config/package.json [hunk 1]
```diff
@@ -10,21 +11,21 @@
...
+      "require": "./dist/tailwind.config.js"
     },
     "./postcss.config": {
-      "default": "./dist/postcss.config.js",
-      "types": "./dist/postcss.config.d.ts"
-    },
-    "./package.json": "./package.json"
+      "types": "./dist/postcss.config.d.ts",
+      "import": "./dist/postcss.config.js",
+      "require": "./dist/postcss.config.js"
```
Comment (dimaMachina) [inline packages/tailwind-config/package.json:23]: storybook complains if there is no `require`

File: packages/tailwind-config/tsup.config.ts [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
+export default defineConfig({
+  name: 'tailwind.config',
+  entry: ['src'],
+  format: 'esm',
+  dts: true,
+  clean: true,
+  bundle: false,
+  plugins: [
+    {
+      // Strip `node:` prefix from imports because Storybook complains about it
```
Comment (hasparus) [inline packages/tailwind-config/tsup.config.ts:12]: turbo annoying

File: .changeset/sixty-hounds-reflect.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+'@theguild/tailwind-config': minor
+---
+
+- use ESM config for `postcss`
+- use `postcss-lightningcss` instead of `autoprefixer` and `cssnano`
```
Comment (hasparus) [inline .changeset/sixty-hounds-reflect.md:6]: TIL about lightningcss

File: .eslintrc.cjs [hunk 4]
```diff
@@ -1,9 +1,10 @@
 module.exports = {
   extends: [
     '@theguild',
+    '@theguild/eslint-config/react',
     '@theguild/eslint-config/json',
     '@theguild/eslint-config/yml',
-    '@theguild/eslint-config/mdx',
+    // '@theguild/eslint-config/mdx',
```
Comment (hasparus) [inline .eslintrc.cjs:7]: It doesn't work now?

File: packages/tailwind-config/tsconfig.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "target": "es2018",
+    "target": "es2021",
```
[the remark this answers is not in the document]
Comment (dimaMachina) [reply packages/tailwind-config/tsconfig.json:3]: I think we don't need even `es2021` is some of states of this PR  I had`createRequire(import.meta.url)` and `tsup` complained about `import.meta.url` being available only in `es2021`

File: .changeset/sixty-hounds-reflect.md (hunk 3 above)
Comment (dimaMachina) [reply .changeset/sixty-hounds-reflect.md:6]: 😁 I used `postcss-lightningcss` in Nextra previously

File: packages/tailwind-config/src/postcss.config.ts [hunk 6]
```diff
@@ -1,13 +1,10 @@
-import { Config } from 'postcss-load-config';
+/* eslint-disable import/no-default-export */
 
-const config: Config = {
+/** @type {import('postcss').Postcss} */
```
[the remark this answers is not in the document]
Comment (dimaMachina) [reply packages/tailwind-config/src/postcss.config.ts:3]: It should, but postcss are broken, I removed it
image

File: .eslintrc.cjs (hunk 4 above)
Comment (dimaMachina) [reply .eslintrc.cjs]: I fixed the broken output!

**Review by hasparus — APPROVED** [review]

**Pull request merged**